components/notebook-controller/Dockerfile

```diff
@@ -10,7 +10,7 @@ ARG SOURCE_CODE=.
 ARG GOLANG_VERSION=1.20
 
 # Use ubi8/go-toolset as base image
-FROM registry.redhat.io/ubi8/go-toolset:${GOLANG_VERSION} as builder
+FROM registry.access.redhat.com/ubi8/go-toolset:${GOLANG_VERSION} as builder
```

Comment:
Why we change the registry here?

Reply:
Because the registry.redhat.io is authenticated and registry.acess.redhat.com does not require auth to pull.

Reply:
Openshift-ci can pull from authenticated registries, but GitHub actions cannot unless I provide pull secret. And I don't want to provide pull secret.

Reply:
Thanks for the information. It looks like we’re set with the upstream builds, but I’m wondering about the downstream side. When this PR gets synced, will we be good to use this registry instead?

Reply:
This dockerfile in the repo is not used for downstream builds. The Dockerfile (Dockerfile.in) for CPaaS builds is held in the gitlab midstream repo and it is updated manually there. Until two weeks ago or so we had in this repository a completely upstream non-rhel Dockerfile https://github.com/opendatahub-io/kubeflow/commit/f0bac824437a71909f380b097efb1280ff2b3823.

Reply:
And yeah, using `registry.redhat.io` in upstream was a mistake I made there. It's fine on OCP-CI which is configured with secret to pull from that, but GitHub Actions cannot.

Reply:
Sound good!